python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -307,12 +307,23 @@ def run(self, splash, version, **kwargs):
         self.startup_descriptor = kwargs.get("startup_descriptor")
         server = kwargs.get("server")
 
-        # Log usage statistics about the Shotgun Desktop executable and the desktop startup.
-        sgtk.util.log_user_attribute_metric("tk-framework-desktopstartup", self.startup_version)
-        sgtk.util.log_user_attribute_metric("Shotgun Desktop version", self.app_version)
-        # If a server is passed down from the desktop startup, it means we won't be using the engine-based
-        # websocket server.
-        sgtk.util.log_user_attribute_metric("Engine-Websockets", "no" if server else "yes")
+        try:
+            # Log usage statistics about the Shotgun Desktop executable and the desktop startup.
+            #
+            # First we update `host_info` property so subsequent metrics can benefit
+            # having the updated information. A special case is made for for Desktop
+            # as we do want both versiond but don't want to create another metric field.
+            # We are then combining both versions into single version string.
+            self._engine._host_info.update({
+                "version": "%s / %s" % (self.app_version, self.startup_version)
+            })
+
+            self._engine.log_metric("Launched Software")
+
+        except Exception as e:
+            logger.exception("Unexpected error logging a metric")
```

Comment:
Should we log the error here?

Reply:
The logger 'exception' method already logs a mini TraceBack, I didn't though it was useful to add anything else. Here is an example:
```
Traceback (most recent call last):
  File "/shotgun/shotgun_workspace/tk-desktop/python/tk_desktop/desktop_engine_site_implementation.py", line 321, in run
    self._engine.log_metdddric("Launched Software")
  File "/shotgun/shotgun_workspace/tk-desktop/engine.py", line 149, in __getattr__
    return getattr(self.__impl, attr)
AttributeError: 'DesktopEngineSiteImplementation' object has no attribute 'log_metdddric'
```

Reply:
Ah, right. Makes sense. All good then.